### PR TITLE
Made new GUI for the orderer

### PR DESCRIPTION
**Pull request by Nefsen402**
This is an experiment. I don't mind if you don't if you don't merge this, besides there are probably some bugs.

Here is the GUI:
mhbcc9e

Clearly I have went for the scroll bar instead of the page layout witch I hate. The scroll bar works like you would normally expect it to, clicking and dragging you mouse drags the scroll bar, pressing the arrow keys, and using the scroll wheel on your mouse moves the scroll bar.

For the search bar, I just used the standard minecraft `GuiTextBox` for this and with that it's more advanced then the previous text box as it lets you select text, move the insertion point, ect. I also have a other text box replacing the item counter on the old GUI, this is the new GUI is the counter as you can type in any number inside the text box, this text box will only accept numbers so typing a letter will do nothing. I'm debating if I should make the focus go to the search bar if I do type a letter in the item count bar; input is appreciated!

You'll also note the request button, it's disabled because there are no items selected.

screenshot_1
This new GUI lets you select multiple items. Pressing control lets you select more then one. When you press the request button, all the items selected will be requested with the item count you specified.

Please let me know what you think about this change! I'd appreciate it if you would give some suggestions to possibly merge this pull request.

### CODE REVIEW

**Comment by davboecki** [comment]
I am currently cleaning up the Orderer code myself, so that it will be easyer to handle in the future. But some of your ideas look good. Maybe I will be able to include some of them in my reorganisation.

**Comment by Nefsen402** [comment]
I have to admit, the old code was a mess, but it was good enough for me to understand it. If people like it, I would clean up the code myself, like redo the request packet since it only supports one request at a time. Anyway, I'm glad you're doing something about this GUI because I hated the old one.

**Comment by GUIpsp** [comment]
What does this new one one have the old one doesn't?

**Comment by Nefsen402** [comment]
~~Are you blind, did you even read the post?~~ I made it clear with screenshots.

Edit by theZorro266: First sentence is totally unnecessary.

**Comment by bziemons** [comment]
From my point of view: You don't have several commits with the same commit message, but instead several commits with different messages and different changes to the system.
Right now it looks like those commits do actually belong together and that means the changes belong in one commit. Luckily Git let's you change that after commiting.

But apart from that, if davboecki is already working on the same code, it would be bad to merge it now anyway. Instead he would take parts of your code and include them in his commits, as he mentioned. That way your support does still find it's way into this project.

**Comment by GUIpsp** [comment]
How does ctrl+a work with multiple selected?

**Comment by Nefsen402** [comment]
It simply sets the item count to the item with the lowest item count out of all the items available. I do notice a small bug with the updating of the item count text box, once I get home from school I'll commit a fix.

File: common/logisticspipes/gui/orderer/FluidGuiOrderer.java [hunk 1]
```diff
@@ -20,20 +24,21 @@ public FluidGuiOrderer(PipeFluidRequestLogistics pipe, EntityPlayer entityPlayer
...
 	}
 	
 	@Override
-	protected void actionPerformed(GuiButton guibutton) {
-		if (guibutton.id == 0 && selectedItem != null){
-			if(editsearch) {
-				editsearchb = false;
+	public void requestItems() {
+		if (requestCount > 0) {
+			Iterator<LoadedItem> iter = this.loadedItems.iterator();
```
Comment (GUIpsp) [inline common/logisticspipes/gui/orderer/FluidGuiOrderer.java]: Why not using a for here?

**Comment by Nefsen402** [comment]
I don't understand?

**Comment by GUIpsp** [comment]
For loops are more readable than using iterators.

**Comment by Nefsen402** [comment]
I guess you could do that but I'm pretty sure the for loops call toArray() and iterating that. It's slower. I might be wrong. I'm using only iterators to guarantee speed because were working with a LinkedList instead if a ArrayList.

**Comment by GUIpsp** [comment]
You are wrong, the for loops work on any iterable, and it uses the list's iterator.

**Comment by Nefsen402** [comment]
You learn something every day! I probably use iterators all the time, now I realize there is no point.

**Comment by Nefsen402** [comment]
I did some more stuff and this time I re-writed some of the networking stuff to support requests of multiple items in one packet. I also did this:

screenshot_3

The chat and this popup thing have all the items you requested in a neat list.

**Comment by bziemons** [comment]
I refuse to even think about pulling this, if there are still many commits without an apropiate commit message. Also pack similar commits together in one commit.

**Comment by Nefsen402** [comment]
Did you know there are such things as branches? Github is great for experimental stuff. If you would be a decent person, you would have created a new branch and pulled this to it. Anyway, none of you people clearly didn't even touch the download button to check it out for yourself.

**Comment by GUIpsp** [comment]
@Need4Speed402 Come back when you have a different, and decent attitude.

**Pull request closed without merging**